sympy/printing/codeprinter.py

```diff
@@ -166,16 +166,17 @@ def doprint(self, expr, assign_to=None):
 
     def _doprint_loops(self, expr, assign_to=None):
         # Here we print an expression that contains Indexed objects, they
-        # correspond to arrays in the generated code.  The low-level implementation
-        # involves looping over array elements and possibly storing results in temporary
-        # variables or accumulate it in the assign_to object.
+        # correspond to arrays in the generated code.  The low-level
+        # implementation involves looping over array elements and possibly
+        # storing results in temporary variables or accumulate it in the
+        # assign_to object.
 
         if self._settings.get('contract', True):
```

Comment:
@bjodah @moorepants @jcrist Yes. How should we handle deprecating it? I'm struggling a bit with this. Perhaps:

Leave default as `contract = True`, and within the `True` branch if it's not an `EinsteinSum` then put out a deprecation warning.

But what about the `False` branch? On the one hand, that means the user is explicitly using the `contract` flag, which we are deprecating, so we should print a deprecation message. On the other hand, the user likely isn't using an `EinsteinSum` in this case, and the `contract = False` logic should be the new default behavior in this case. So we shouldn't print a warning?